pkg/machinepool/machinepool.go

```diff
@@ -0,0 +1,121 @@
+package machinepool
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+
+	"github.com/openshift/rosa/pkg/output"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+var fetchMessage string = "Fetching %s '%s' for cluster '%s'"
+var notFoundMessage string = "Machine pool '%s' not found"
+
+//go:generate mockgen -source=machinepool.go -package=mocks -destination=machinepool_mock.go
+type MachinePoolService interface {
+	DescribeMachinePool(*rosa.Runtime, *cmv1.Cluster, string, string) error
```

Comment:
I believe it is possible to have a single describe as they both take the same arguments, a factory to differentiate based on cluster topology

Reply:
Implemented a factory-style solution, and kept it simple. This removed a few lines from the command in the end, which is a nice outcome.